x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/lib/is_rate.ts

```diff
@@ -32,12 +32,12 @@ export const isCustomMetricRate = (customMetric: SnapshotCustomMetricInput) => {
 };
 
 export const isInterfaceRateAgg = (metric: MetricsUIAggregation | undefined) => {
-  if (!MetricsUIAggregationRT.is(metric)) {
+  if (hasAggregations(metric) || metric === undefined) {
     return false;
   }
   const values = Object.values(metric);
   return (
-    values.some((agg) => ESTermsWithAggregationRT.is(agg)) &&
+    values.some((agg) => hasSnapshotTermsWithAggregation(agg)) &&
```

Comment:
Wdyt if we keep `ESTermsWithAggregationRT` for `alerting` stuff?

Reply:
I was trying that but the issue I faced is that `ESTermsWithAggregationRT` is still using the aggregation type we replace and if we try to still use it then `SnapshotTermsWithAggregation`  is using already the new `MetricsUIAggregation` which depends on the `estypes` (`Record<string, estypes.AggregationsAggregate>`) and there will be incompatible with the "old" type. 
I will think about another way to keep them compatible but so far what I tried was not working

Reply:
So I used the old es types and `SnapshotTermsWithAggregation` renamed to `SnapshotTermsWithAggregationForAlerts` to keep them the same as before but this will also need refactoring in the future I guess